core/arch/arm/kernel/generic_boot.c

```diff
@@ -753,22 +754,40 @@ paddr_t generic_boot_core_hpen(void)
  * bits from e-fuses or whatever.
  *
  * The default implementation just sets it to a constant.
+ *
+ * If your bootloader passes OP-TEE a DTB, then if it has a property
+ * firmware/optee-pseudo-device-id the data found in there is
+ * used to generate a pseudo id.  This may have been sourced from,
+ * eg, eMMC CID serial number that is specific to the individual
+ * PCB, if not the SoC.
  */
 
 __weak void tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 {
-	memset(&hwkey->data[0], 0, sizeof(hwkey->data));
+	const uint8_t *p = (uint8_t *)&pseudo_device_id;
+	size_t n;
+
+	if (!pseudo_device_id_len) {
+		memset(&hwkey->data[0], 0, sizeof(hwkey->data));
+		return;
+	}
+
+	for (n = 0; n < sizeof(hwkey->data); n++)
+		hwkey->data[n] = p[n % pseudo_device_id_len] ^ 0xff;
```

Comment:
I don't know.  I thought it might be good if the "cpu id" and the hwkey data were not always identical.  They are no treated as identical in the current case (hwkey is all 0 and "cpu id" is all beefy.